students/KyleBarry/session08/sparse_array.py

```diff
@@ -0,0 +1,44 @@
+"""Emulate a list in a custom class"""
+
+
+class SparseArray:
+    """Create array objects that act as lists"""
+
+    def __init__(self, seq):
+        """Establish variable for sequence argument"""
+        self.seq = seq
```

Comment:
this isn't going to give you any sparse behaviour -- you need to store the non-zero values some other way.

Also if you want to store stuff in a sequence, you probaby want to know exactly what type it is here -- for instance if someone init's with a tuple, then any of the mutating methods wouldn't work. So at least:

`self.seq = list(seq)`

which isn't a horrible way to develop and test you magic methods, and then you can make it spase later -- but he sparse part is the tricky part :-)